src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Streaming.cs

```diff
@@ -80,6 +85,10 @@ public async Task SendStreamingUpdatesAsync(HttpContext httpContext, Task untilT
             HandleExceptionAfterResponseStarted(_httpContext, writer, ex);
             await writer.FlushAsync(); // Important otherwise the client won't receive the error message, as we're about to fail the pipeline
             await _httpContext.Response.CompleteAsync();
+
+            // We need to inform the debugger that this exception should be considered user unhandled unlike the NavigationException.
+            // Review: Is this necessary if the method attributed with [DebuggerDisableUserUnhandledExceptions] rethrows like this does?
+            Debugger.BreakForUserUnhandledException(ex);
```

Comment:
@gregg-miskelly Is this necessary given that the method rethrows?

Reply:
That is a good question. The VS Debugger's current implementation is that we will wind up eventually breaking as long as some other part of non-user code catches that exception. I *think* that behavior sounds right to me, but I could be convinced in either direction. Do you have an opinion?

CC @asundheim

Reply:
It's probably for the best if we break as close to the original user-code throw as possible, in case the HttpContext is altered and the user wants to inspect it in as close to a state as it was when they threw? (in which case, maybe this should be the first line of the `catch`?)

To the question, it looks like we actually toggle the notifications for the exception off once we get CatchHandlerFound in a non-user frame, regardless if the frame has the `Disable` attribute, maybe that should be changed on our end? My original thought was that if one frame says we shouldn't break, then we shouldn't make every single frame above it add the attribute if it rethrew through a couple of frames before landing where it needed to be caught, but it looks like that's not the case?

If we don't change it, then this is necessary, if not, it becomes this method's callers problem, but we'll not get a chance to break until the caller catches it (implicitly or otherwise)

Reply:
I'll move this up to the start of the `catch (Exception ex)` block since it sounds like it doesn't hurt regardless of how we decide the debugger should handle rethrows. I like that this would avoid mutating the HttpContext before the debugger breaks as you point out.

This would only be problematic if the debugger stopped both for the `Debugger.BreakForUserUnhandledException(ex)` and later after the rethrow, but it sounds like that would never happen.

> My original thought was that if one frame says we shouldn't break, then we shouldn't make every single frame above it add the attribute if it rethrew through a couple of frames before landing where it needed to be caught, but it looks like that's not the case?

My intuition was that `[DebuggerDisableUserUnhandledExceptions]` would only affect exceptions that were caught and swallowed by the attributed method. I think the method either not catching an exception or rethrowing it is a pretty clear indication that the exception should still be considered "unhandled".

If there's an outer method that is catching and handling these rethrown exceptions, I would expect to be required to use `[DebuggerDisableUserUnhandledExceptions]` on that method too. Or more likely, I would only attribute the outermost method call that does the catching.